packages/smart-wallet/src/smartWallet.js

```diff
@@ -282,14 +282,13 @@ export const prepareSmartWallet = (baggage, shared) => {
         /**
          * @param {RemotePurse} purse
          * @param {Amount<any>} balance
-         * @param {'init'} [init]
          */
-        updateBalance(purse, balance, init) {
+        updateBalance(purse, balance) {
           const { purseBalances, updatePublishKit } = this.state;
-          if (init) {
-            purseBalances.init(purse, balance);
-          } else {
+          if (purseBalances.has(purse)) {
             purseBalances.set(purse, balance);
+          } else {
+            purseBalances.init(purse, balance);
```

Comment:
This looks like the gist of it. I'd appreciate a look by @mhofman and/or @gibson042 .

Did @gibson042 already look at this?

Reply:
This looks fine. Unfortunate we don't have an `upsert` for these cases.